src/context/modal-context.tsx

```diff
@@ -31,12 +31,14 @@ const ModalProvider: FC<ModalProviderProps> = ({ children }) => {
   const [modal, setModal] = useState<React.ReactElement | null>(null)
   const [paperProps, setPaperProps] = useState<PaperProps>({})
   const [timer, setTimer] = useState<NodeJS.Timeout | null>(null)
+  const [closeFunc, setCloseFunc] = useState<() => void>(() => console.log())
```

Comment:
can you pass onClose function as a props to this context and then if it exists use it in closeModal function?

Reply:
not relevant